crates/stackable-operator/src/commons/rbac.rs

```diff
@@ -30,12 +30,18 @@ pub enum Error {
 /// Build RBAC objects for the product workloads.
 /// The `rbac_prefix` is meant to be the product name, for example: zookeeper, airflow, etc.
 /// and it is a assumed that a ClusterRole named `{rbac_prefix}-clusterrole` exists.
+/// 'rbac_prefix' is not used to build the names of the serviceAccount and roleBinding objects,
+/// as this caused problems with multiple clusters of the same product within the same namespace
+/// see <https://stackable.atlassian.net/browse/SUP-148> for more details.
+/// Instead the names for these objects are created by reading the name from the cluster object
+/// and appending [-rolebinding|-serviceaccount] to create unique names instead of using the
+/// same objects for multiple clusters.
 pub fn build_rbac_resources<T: Clone + Resource<DynamicType = ()>>(
     resource: &T,
     rbac_prefix: &str,
     labels: Labels,
 ) -> Result<(ServiceAccount, RoleBinding)> {
-    let sa_name = service_account_name(rbac_prefix);
+    let sa_name = service_account_name(&resource.name_any());
```

Comment:
The rolebinding should also cover the old name for at least one release, or we'll have downtime during the switchover.

Reply:
(Since existing pods will have the old ServiceAccount name, which will no longer be bound to any roles after this change goes through.)

Reply:
changed it. We probably want to track that on our deprecation board?

Reply:
Absolutely.

Reply:
We are just discussing if this is really necessary.. since we delete the old serviceAccount pretty much immediately, won't the double roleBinding be ineffective anyway?

Reply:
It's worth testing, but AFAIK deleting the old ServiceAccount will only prevent new Pods from being spawned that use it. Any existing pods would still have a valid authn token for the old SA, and their requests still need to be authorized correctly until they're migrated over to the new SA. We're still talking about a relatively short window, but on the order of minutes, not seconds.

However, it gets slightly worse. StatefulSet upgrades will only progress if the STS replicas are in a healthy state, so services that require ongoing K8s access would effectively get stuck in a broken state until it is taken down manually.

I'm not sure how many of our products actually require K8s access these days, but I suspect at least NiFi would be a likely problem case?